packages/amplify_api/lib/graphql/graphql_helpers.dart

```diff
@@ -0,0 +1,155 @@
+ */
+
+import 'package:amplify_api_plugin_interface/amplify_api_plugin_interface.dart';
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:flutter/foundation.dart';
+import '../amplify_api.dart';
+
+class ModelQueriesFactory extends ModelQueriesInterface {
+  // Singleton methods/properties
+  // usage: ModelQueriesFactory.instance;
+  ModelQueriesFactory._();
+
+  static final ModelQueriesFactory _instance = ModelQueriesFactory._();
+
+  static ModelQueriesFactory get instance => _instance;
+
+  @override
+  GraphQLRequest<T> get<T extends Model>(ModelType<T> modelType, String id) {
+    Map<String, String> variableInput = {"id": "ID"};
+
+    return GraphQLRequestFactory().buildQuery<T>(
+        modelType: modelType,
+        variableInput: variableInput,
+        id: id,
+        requestType: GraphQLRequestType.query,
+        requestOperation: GraphQLRequestOperation.get);
+  }
+}
+
+class GraphQLRequestFactory extends GraphQLRequestFactoryInterface {
+  String _getModelType(ModelFieldTypeEnum? val) {
+    switch (val) {
+      case ModelFieldTypeEnum.string:
+        return "String";
+      case ModelFieldTypeEnum.int:
+        return "int";
+      case ModelFieldTypeEnum.model:
+        return "ID";
+      default:
+        return "Error: UnknownType!";
+    }
+  }
+
+  String _getFieldsFromModelType(ModelSchema schema) {
+    Map<String, ModelField?>? fieldsMap = schema.fields;
+    return fieldsMap?.entries
+            .map((entry) => entry.value?.association == null ? entry.key : '')
+            .toList()
+            .join(' ') ??
+        '';
+  }
+
+  ModelSchema _getSchema(ModelType modelType) {
+    ModelProviderInterface? provider = AmplifyAPI.instance.getModelProvider();
+
+    if (provider == null) {
+      throw ApiException('No modelProvider found',
+          recoverySuggestion:
+              'Pass in a modelProvider instance while instantiating APIPlugin');
+    }
+
+    ModelSchema schema = provider.modelSchemas.firstWhere(
+        (elem) => provider.getModelTypeByModelName(elem.name) == modelType);
+
+    if (schema == null) {
+      throw ApiException('No schema found for the ModelType provided',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a vaild ModelType');
+    }
+
+    return schema;
+  }
+
+  /*
+   *  buildQuery()
+   *
+   *  
+   *  Example: 
+   *    query getBlog($id: ID!) { getBlog(id: $id) { id name createdAt } }
+  */
+  @override
+  GraphQLRequest<T> buildQuery<T extends Model>(
+      {required ModelType modelType,
+      required Map<String, String>? variableInput,
+      required String? id,
+      required GraphQLRequestType requestType,
+      required GraphQLRequestOperation requestOperation}) {
+    ModelSchema schema = _getSchema(modelType);
+
+    String? name = requestOperation == GraphQLRequestOperation.list
```

Comment:
nit: I suggest declaring a boolean variable like `isListOperation` above and then using that variable in this if statement, as well as same condition repeated below. That way, this condition is only defined once. Another thing to think about is if all the list-specific conditions can be in the same place? Maybe not, but might make it clearer for reader to know what happens if list.

Reply:
Fair enough, I consolidated logic to avoid multiple conditions. Previously I was trying to have everything only defined once, but this seems like a better flow.